pkg/controller/kubelet-config/kubelet_config_controller.go

```diff
@@ -621,7 +620,14 @@ func (ctrl *Controller) syncKubeletConfig(key string) error {
 			return ctrl.syncStatusOnly(cfg, err, "could not find MachineConfig: %v", managedKey)
 		}
 		isNotFound := macherrors.IsNotFound(err)
-
+		// If we have seen this generation and the sync didn't fail, then skip
+		if !isNotFound && cfg.Status.ObservedGeneration >= cfg.Generation && cfg.Status.Conditions[len(cfg.Status.Conditions)-1].Type == mcfgv1.KubeletConfigSuccess {
```

Comment:
Hmm, so I know we do the `[len(cfg.Status.Conditions)-1]` elsewhere, I assume the last condition in the list is always the latest? (hence success here)

Reply:
yup follows the similar pattern as ContainerRuntimeConfig